nuxeo-services/login/nuxeo-platform-login-cas2/src/main/java/org/nuxeo/ecm/platform/ui/web/auth/cas2/SecurityExceptionHandler.java

```diff
@@ -67,9 +71,18 @@ public void handleException(HttpServletRequest request, HttpServletResponse resp
         }
 
         Principal principal = request.getUserPrincipal();
-        NuxeoPrincipal nuxeoPrincipal = null;
+        if (principal == null) {
+            LoginContext loginContext = (LoginContext) request.getAttribute(LOGINCONTEXT_KEY);
+            Subject subject = loginContext.getSubject();
+            if (subject != null) {
+                Set<Principal> principals = subject.getPrincipals();
+                if (principals != null && !principals.isEmpty()) {
+                    principal = (Principal) principals.toArray()[0];
```

Comment:
Note for later, you can do this a bit more elegantly with: `principals.iterator().next()`